### PR TITLE
Add TypeScript typings

### PR DESCRIPTION
**Pull request by resynth1943**
Hey there! I don't use this library, but I noticed people were asking for TypeScript typings: #22, so I thought I would help out and add them. If there's anything you're not happy with, hit me up :-)

### CODE REVIEW

**Comment by resynth1943** [comment]
@googlebot I signed it!

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		method?: TMethods;
+
+		/** Requwst headers */
+		headers?: Headers;
+
+		/** Authorization header value to send with the request */
+		auth?: string;
+
+		/** Pass an Cross-site Request Forgery prevention cookie value as a header defined by `xsrfHeaderName` */
+		ssrfCookieName?: string;
```
Comment (developit) [inline index.d.ts:66]:
```suggestion
		xsrfCookieName?: string;
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+		/** The name of a header to use for passing XSRF cookies */
+		ssrfHeaderName?: string;
```
Comment (developit) [inline index.d.ts:69]:
```suggestion
		xsrfHeaderName?: string;
```

**Comment by developit** [comment]
@resynth1943 - these look good! I'm wondering if it would be possible to somehow re-export Axios's original typings though? That would mean folks could move between the two libraries without causing any type errors.

File: index.d.ts (hunk 1 above)
Comment (resynth1943) [reply index.d.ts:66]: Goddammit.

File: index.d.ts (hunk 2 above)
Comment (resynth1943) [reply index.d.ts:69]: *WHY*

**Comment by resynth1943** [comment]
> these look good! I'm wondering if it would be possible to somehow re-export Axios's original typings though? That would mean folks could move between the two libraries without causing any type errors.
Hmm, not sure... I'll try and look into it.

**Comment by resynth1943** [comment]
Oh my god, not this again.

**Comment by RyanChristian4427** [comment]
@googlebot I fixed it.

**Comment by developit** [comment]
Hiya @resynth1943! This looks pretty nice, and it's clear there's more type safety possible by moving the types into a dedicated declaration file like you have done here. Any ideas for how we could combine this with the JSDoc tweaks from #22?

**Comment by resynth1943** [comment]
Oh dang, I've never had to maintain JSDoc alongside TypeScript (I personally dislike JSDoc). I can look into it though, if you want ...?

**Comment by gianpaj** [comment]
could somebody by any chance add TS types to this repo or create a `@types/redaxios`? 🙏

**Comment by resynth1943** [comment]
Oh shit this is still open?

**Comment by samwhiteiv** [comment]
possible to merge this? would be greatly appreciated!

also, could somebody please add TS types to this repo or create a @types/redaxios? thank you

**Comment by rschristian** [comment]
@samwhiteiv Types are already shipped, and have been for quite a while. They're just generated from JSDoc rather than authored directly. See #33

**Comment by samwhiteiv** [comment]
noted. thanks @rschristian

**Comment by developit** [comment]
Yep, sorry about that. There might still be some improvements to be had in the JSDoc types though.

**Comment by rschristian** [comment]
> Well, it seems there is no update on this issue, so bad.

bad?

> I can't get `import axios, { AxiosInstance, AxiosRequestConfig } from 'redaxios';` working:

That's correct behavior for now, as neither type exists. Mind creating a new issue for adding these types? Comments in a closed PR aren't great for tracking potential hiccups for users.

**Pull request closed without merging**